packages/documentation/.storybook/main.ts

```diff
@@ -1,7 +1,7 @@
 import type { StorybookConfig } from '@storybook/web-components-vite';
 import pkg from '../package.json';
-import remarkAutolinkHeadings from 'remark-autolink-headings';
 import { mergeConfig } from 'vite';
+import rehypeAutolinkHeadings from 'rehype-autolink-headings';
```

Comment:
I had another try and I managed to make it work with this plugin which replace the legacy `remark-autolink-headings` (see https://github.com/remarkjs/remark-autolink-headings)